python/fusion_engine_client/messages/defs.py

```diff
@@ -402,12 +317,44 @@ def get_type_string(cls):
     def get_version(cls) -> int:
         return cls.MESSAGE_VERSION
 
+    @classmethod
+    def is_instance(cls, obj) -> bool:
```

Comment:
I'm confused what the use of these functions is. Is it so you can call:
`MessagePayload.is_instance(obj)`
instead of 
`isinstance(obj, MessagePayload)` ?

Reply:
Yeah, there's honestly no great reason for it except for consistency. Ideally we wouldn't have either of these functions, and would just call `isinstance(obj, Foo)` and `issubclass(obj, Foo)`. Turns out though that `isinstance(obj, Foo)` works for any value of `obj` (instance, some other type, `None`, etc.), but `issubclass(obj, Foo)` throws a `TypeError` if `obj` is not a class.

So I made this `is_subclass()` helper strictly for convenience since we're now doing that check in a few places within a larger if/else statement. I didn't want all of those places to have to know how to do the right thing and call `inspect.isclass()` since that was getting ugly. Then once I added `is_subclass()` I figured I should add `is_instance()` for completeness.

I can remove it if you think it's confusing though.

Reply:
Not really a big deal either way, I'd say remove it since it makes it clearer that the `is_subclass` is meant to fix that specific issue and there's not another reason for making the class members.

Reply:
Ok I removed `is_instance()`, and documented `is_subclass()` as follows:
```py
    @classmethod
    def is_subclass(cls, obj) -> bool:
        """!
        @brief Check if an object is a _class_, which is derived from @ref MessagePayload.

        This function calls the built-in `issubclass()` operator to check if a specified object is a class derived from
        @ref MessagePayload. Unlike the operator, which raises a `TypeError` if the object is not a class (e.g., if you
        pass in a `None`, or any other object), this function accepts any type for its argument:
        ```py
        issubclass(None, MessagePayload)        # TypeError
        issubclass(float, MessagePayload)       # False
        issubclass(dict, MessagePayload)        # False
        issubclass(PoseMessage, MessagePayload) # True

        MessagePayload.is_subclass(None)        # False
        MessagePayload.is_subclass(float)       # False
        MessagePayload.is_subclass(dict)        # False
        MessagePayload.is_subclass(PoseMessage) # True
        ```

        Note that this function is specifically meant to check classes, not class _instances_. To test if an object is
        an instance of a class derived from @ref MessagePayload, use the `isinstance()` operator instead:
        ```py
        isinstance(None, MessagePayload)          # False
        isinstance(3.6, MessagePayload)           # False
        isinstance(dict(), MessagePayload)        # False
        isinstance(PoseMessage(), MessagePayload) # True
        ```

        @param obj The object to be tested.
        """
```